atlp-rwanda/e-comm-team-amigos-bn
# #184581180 Update User Profile

#### What does this PR do?

This PR allow user to get and update his profile. It also allow other authenticated users to get a certain user's profile

#### Description of Task to be completed?

Have the following endpoints working

- `GET /user/profile`
- `GET /user/<user_id>`
- `PATCH /user/updateMe` 

#### How should this be manually tested?

First make sure you have docker installed. In your terminal;

1. In your **e-comm-team-amigos-bn** directory, open terminal and run `git fetch origin pull/28/ft-user-profile-184581180`
2. Checkout to **ft-user-profile-184581180** branch
3. Run `npm install`
4. Run `npm run dev`
5. In postman or documentation do the following
     - Create Account / Signup
     - Use the returned token for authorizarion
     - Make requests on above mentioned endpoints.

#### Any background context you want to provide?

- Have Node Js installed in your local machine
- Should have express server running
- Should have your `.env` file with neccessary variables as in `.env.example`
     
#### What are the relevant pivotal tracker/Trello stories?

Story ID: [184581180](https://www.pivotaltracker.com/story/show/184581180)
